pkg/es/config/config_test.go

```diff
@@ -0,0 +1,605 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+)
+
+var mockEsServerResponseWithVersion0 = []byte(`
+{
+	"Version": {
+		"Number": "0"
+	}
+}
+`)
+
+var mockEsServerResponseWithVersion1 = []byte(`
+{
+	"tagline": "OpenSearch",
+	"Version": {
+		"Number": "1"
+	}
+}
+`)
+
+var mockEsServerResponseWithVersion2 = []byte(`
+{
+	"tagline": "OpenSearch",
+	"Version": {
+		"Number": "2"
+	}
+}
+`)
+
+func copyToTempFile(t *testing.T, pattern string, filename string) (file *os.File, closeFn func()) {
+	tempFile, err := os.CreateTemp("", pattern+"_")
```

Comment:
You should use `t.TempDir`, then the directory will be automatically removed and you won't need to return a cleanup function. 

In case you do need to perform some clean-up (e.g. in some future test), rather than returning a function you can register it via `t.Cleanup(fn)`, which also requires less boilerplate code in the tests.